Sources/MongoSwift/SDAM.swift

```diff
@@ -167,9 +162,35 @@ public struct ServerDescription {
         self.logicalSessionTimeoutMinutes = isMaster["logicalSessionTimeoutMinutes"] as? Int64
     }
 
+    /// An internal function used to determine if a given server matches a tag set.
+    internal func matchesTagSet(_ tagSet: Document) -> Bool {
+        guard let serverTags = self.tags else {
+            return false
+        }
+
+        for kvp in tagSet {
+            if !serverTags.keys.contains(kvp.key) || !bsonEquals(serverTags[kvp.key]!, kvp.value) {
+                return false
+            }
+        }
+
+        return true
+    }
+
+    /// An internal initializer to create a `ServerDescription` for testing purposes.
+    internal init(_ connectionId: ConnectionId, _ rtt: Int64, _ isMaster: Document, _ type: ServerType,
```

Comment:
1. Done
2. You can only define convenience initializers in class extensions, so moving this initializer to the tests package won't work unfortunately

Reply:
ugh! ok, that's fine. let's just add an extra line to the docstring here explaining why this test initializer lives here, because otherwise we will probably forget and try to move them again in the future :)

Reply:
but hmm, wait. we used to have an initializer in [this extension of `Document`](https://github.com/mongodb/mongo-swift-driver/blob/aeb8e390640a65c1f1a7be8ad4e67b4cecabe76a/Sources/MongoSwift/BSON/Document.swift#L380).
perhaps it's something about defining it within a different package?

Reply:
Oh, you're right!
https://docs.swift.org/swift-book/LanguageGuide/Extensions.html#ID153 says explicitly for _classes_, not structs/value types, which `Document`, `TopologyDescription`, and `ServerDescription` all are.

But yes, I can't put the initializers in the test file due to it being in a separate package. I can move them around within MongoSwift though.

Reply:
So after some more digging, what we want actually _is_ possible in Swift 4, but the compiler will give a warning. In Swift 5, that warning will be an error (https://github.com/apple/swift-evolution/blob/master/proposals/0189-restrict-cross-module-struct-initializers.md). 

The reason all this confusion started is I read the aforementioned warning as an error and skimmed over the actual, un/kinda related one in the build output. 

What I'm not entirely clear on is why our tests can access functions marked `internal`, yet are considered outside of the module when implementing the initializers. Are our tests in some half-in-half-out state?

Reply:
ah, I see. if it's going to be forbidden in Swift 5, then I don't think there is a point in doing it now either. 
and ha, we actually had a discussion about this in #swift last week. the reason we have that capability is in the import statement: note that our tests do `@testable import MongoSwift`, which elevates our access privileges.

from [docs](https://docs.swift.org/swift-book/LanguageGuide/AccessControl.html):
>  When you write an app with a unit test target, the code in your app needs to be made available to that module in order to be tested. By default, only entities marked as open or public are accessible to other modules. However, a unit test target can access any internal entity, if you mark the import declaration for a product module with the @testable attribute and compile that product module with testing enabled.

[the last section](https://github.com/apple/swift-evolution/blob/master/proposals/0189-restrict-cross-module-struct-initializers.md#add-an-exception-for-unit-tests) of the proposal you linked notes that they considered making an exception and allowing this when the module is imported via `@testable`, but that they ultimately chose not to since it can already be accomplished in exactly the way you've done it here. 

so yeah, it sounds like we should just leave them where they are, and briefly note why they're there in docstring.

Reply:
(I would also explicitly mention SE-0189)

Reply:
Of course the one paragraph of that proposal I skipped was the most relevant...
Anyways, sounds good. Ill update the docstrings.